app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -655,10 +655,11 @@ private boolean processReturnFromFormEntry(int resultCode, Intent intent) {
 
             // The form is either ready for processing, or not, depending on how it was saved
             if (complete) {
-                // We're honoring in order submissions, now, so trigger a full
-                // submission cycle
+                // Now that we know this form is completed, we can give it the next available
+                // submission ordering number
+                current.setFormNumberForSubmissionOrdering();
```

Comment:
@ctsims Hey, so I just switched to this strategy in the latest commit, but I'm a little hesitant about it. Agreed that it's definitely a cleaner way to make sure we're always using the right value, but it comes at the cost of making each call to `setFormNumberForSubmissionOrdering()` more expensive, since we're now iterating through all the form records in the db each time. I'm holding off on merging this just to hear your thoughts/confirmation on that being a worthwhile trade-off.